### PR TITLE
PLATY-131: Added delete functionality to remove infected files automa…

### PR DESCRIPTION
**Pull request by JoPintoPaul**
…tically.

### CODE REVIEW

File: app/uk/gov/hmrc/fileupload/s3/S3JavaSdkService.scala [hunk 1]
```diff
@@ -211,6 +211,17 @@ class S3JavaSdkService(configuration: com.typesafe.config.Config, metrics: Metri
...
     )
   }
+
+  def deleteObjectFromBucket(bucketName: String, key: String): Unit = {
+    Try(s3Client.deleteObject(bucketName, key)) match {
+      case Success(_) =>
+        Logger.info(s"Objected successfully deleted with key $key from bucket $bucketName")
+        ()
+      case Failure(error) =>
+        Logger.error(s"Attempted to delete object with key $key from bucket $bucketName but error thrown: ${error.getMessage}")
```
Comment (tomasz-rosiek) [inline app/uk/gov/hmrc/fileupload/s3/S3JavaSdkService.scala]: There is a tricky thing about our kibana dashboard - only log entries with exceptions are being shown in 'service exceptions' section.
Could we also log the error then ? (i mean adding error as a last parameter of Logger.error())

File: app/uk/gov/hmrc/fileupload/s3/S3Service.scala [hunk 2]
```diff
@@ -73,6 +77,8 @@ object S3Service {
   type UploadToQuarantine = (String, InputStream, Int) => Future[UploadResult]
 
   type DownloadFromBucket = (S3KeyName) => Option[StreamWithMetadata]
+
+  type DeleteFileFromQuarantineBucket = (String) => Unit
```
Comment (tomasz-rosiek) [inline app/uk/gov/hmrc/fileupload/s3/S3Service.scala]: Shouldn't it have the type S3KeyName ?

File: app/uk/gov/hmrc/fileupload/virusscan/ScannerActor.scala [hunk 3]
```diff
@@ -68,6 +71,7 @@ class ScannerActor(subscribe: (ActorRef, Class[_]) => Boolean,
           scanNext()
         case Xor.Left(ScanResultVirusDetected) =>
           notify(hasVirus = true)
+          deleteInfectedFile()
```
Comment (tomasz-rosiek) [inline app/uk/gov/hmrc/fileupload/virusscan/ScannerActor.scala]: I wonder if we should directly and synchronously delete the file in this actor, or we should have separate actor that reacts to "MarkFileAsInfected" event.
(as we do for clean files - in TransferAgent")

The only difference I see is that with actor approach we don't have to wait until the file is deleted.

File: test/uk/gov/hmrc/fileupload/virusscan/ScannerActorSpec.scala [hunk 4]
```diff
@@ -24,34 +24,60 @@ import org.scalatest.concurrent.Eventually
...
 
       eventually {
-        collector shouldBe expectedCollector(newEventToSend)
+        collector shouldBe expectedCleanCollector(newEventToSend)
+      }
+    }
+
+    "scan files, log when viruses are detected, and delete infected files" in new ScanFixture {
+      val delete = mock[DeleteFileFromQuarantineBucket]
+      (delete.apply(_: String)).expects(*).returns(()).repeat(6 to 6)
```
Comment (tomasz-rosiek) [inline test/uk/gov/hmrc/fileupload/virusscan/ScannerActorSpec.scala]: What 6 to 6 means?

File: app/uk/gov/hmrc/fileupload/s3/S3Service.scala (hunk 2 above)
Comment (JoPintoPaul) [reply app/uk/gov/hmrc/fileupload/s3/S3Service.scala]: As discussed IRL, will leave this as string as the value class isn't widely used in the codebase.

File: app/uk/gov/hmrc/fileupload/s3/S3JavaSdkService.scala (hunk 1 above)
Comment (JoPintoPaul) [reply app/uk/gov/hmrc/fileupload/s3/S3JavaSdkService.scala]: Change pushed.

File: test/uk/gov/hmrc/fileupload/virusscan/ScannerActorSpec.scala [hunk 5]
```diff
@@ -25,23 +25,22 @@ import org.scalatest.time.{Seconds, Span}
 import play.api.libs.json.Json
 import uk.gov.hmrc.fileupload.notifier.NotifierService.NotifySuccess
 import uk.gov.hmrc.fileupload.notifier.{CommandHandler, MarkFileAsClean, MarkFileAsInfected, QuarantineFile}
-import uk.gov.hmrc.fileupload.s3.S3Service.DeleteFileFromQuarantineBucket
 import uk.gov.hmrc.fileupload.virusscan.ScanningService.{ScanResult, ScanResultFileClean, ScanResultVirusDetected}
 import uk.gov.hmrc.fileupload.{EnvelopeId, FileId, FileRefId, StopSystemAfterAll}
 import uk.gov.hmrc.play.test.UnitSpec
-import org.scalamock.scalatest.MockFactory
+//import org.scalamock.scalatest.MockFactory
```
Comment (tomasz-rosiek) [inline test/uk/gov/hmrc/fileupload/virusscan/ScannerActorSpec.scala]: Line to remove :)

**Pull request merged**